engine/access/state_stream/engine.go

```diff
@@ -151,7 +117,6 @@ func NewEng(
 	}
 
 	e.ComponentManager = component.NewComponentManagerBuilder().
```

Comment:
Details [here](https://github.com/onflow/flow-go/pull/4411#discussion_r1223166014)

Reply:
As is, the engine's `component.Component` interface will indicate to the node that it started and immediately shutdown because there are no workers running. At a minimum this will cause weird issues in tests, but may also cause hard to debug issues at runtime.

This suggestion is about attaching the engine's lifecycle to the server that provides it with data.

Reply:
bump